From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Clear out content for the signature we're building
Bug-Ubuntu: https://launchpad.net/bugs/1526959
Last-Update: 2015-12-17

OpenSSL 1.0.2e now actively checks for both data and contents being present
for a certificate. Clear out contents so that we have only data, and run a
chance of actually verifying the signature.

---
 src/sbverify.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: b/src/sbverify.c
===================================================================
--- a/src/sbverify.c
+++ b/src/sbverify.c
@@ -324,6 +324,10 @@ int main(int argc, char **argv)
 	if (!verify)
 		flags |= PKCS7_NOVERIFY;
 
+	/* OpenSSL 1.0.2e no longer allows calling PKCS7_verify with
+	 * both data and content. Empty out the content. */
+	p7->d.sign->contents->d.ptr=0;
+
 	X509_STORE_set_verify_cb_func(certs, x509_verify_cb);
 	rc = PKCS7_verify(p7, NULL, certs, idcbio, NULL, flags);
 	if (!rc) {
